DOCS-11470 comprehensive publication (#141)
mulesoft/docs-exchange: modules/ROOT/pages/exchange-api.adoc

|type |The type of asset to be created. |Required if the type cannot be inferred from the file extension
|dependencies |A string of comma-separated dependencies, defined as `dependencies=GROUP_ID:ASSET_ID:VERSION,GROUP_ID:ASSET_ID:VERSION`. |Optional
|keywords |A string of comma-separated values to be saved as an asset's attributes, for example, `keywords=api,raml,auth,jwt,token`. Search queries can find assets by keyword. |Optional

> |tags |A string of comma-separated values to post as the asset's tags, for example, `tags=tag1,tag2`. If the tags field is not present or is empty in the request, the asset's tags remain unchanged. The string is the final version, so to delete one tag from an asset with two tags, send only the tag that will remain. |Optional

Reviewer: ```suggestion
|tags |A string of comma-separated values to post as the asset's tags, for example, `tags=tag1,tag2`. If the tags field is not present in the request or is empty, the asset's tags remain unchanged. The string is the final version, so to delete one tag from an asset with two tags and send only the tag that you want to remain. |Optional
```

Author: Fixed.

> |fields.key |A string of comma-separated values to post as the asset's custom fields. Replace `key` with the key of the custom fields that you want to add or replace. The custom field must be configured in your organization. If the fields element is not present or is empty in the request, the asset's fields remain unchanged. The string is the final version, so to delete one custom field from an asset with two custom fields, send only the field that will remain. |Optional

Reviewer: ```suggestion
|fields.key |A string of comma-separated values to post as the asset's custom fields. Replace `key` with the key of the custom fields that you want to add or replace. The custom field must be configured in your organization. If the fields element is not present in the request or is empty, the asset's fields remain unchanged. The string is the final version, so to delete one custom field from an asset with two custom fields and send only the field that you want to remain. |Optional
```

Author: Fixed.

> |categories.key |A string of comma-separated values to post as the asset's categories. Replace `key` with the key of the category that you want to add or replace. The category must be configured in your organization. If the categories field is not present or is empty in the request, the asset's categories remain unchanged. The string is the final version, so to delete one category from an asset with two categories, send only the category that will remain. |Optional

Reviewer: ditto ^

Author: Fixed.

The endpoint `publicationStatusLink` provides the asset publication status.

[[set-mutable-data]]
=== Set Mutable Data

This example shows how to create your asset and set the mutable data describing it in the same request. The mutable data of an asset includes categories, custom fields, tags, and documentation pages.

> This example publishes the category and custom field, which have the keys `categoryKey` and `fieldKey`. They must be configured for your organization. Publication fails if they are not configured for your organization or if they are not deleted from the pom.

Reviewer 2: ```suggestion
This example publishes the category and custom field, which have the keys `categoryKey` and `fieldKey`. They must be configured for your organization. Publication fails if they are not configured for your organization.
```

Author: Fixed.

> This endpoint creates a new asset. If the asset already exists, the request fails with error 409 (conflict). Do not try to change the mutable data of an existing version. Instead, create a new patch version with the updated mutable data. For example, if the current version is 1.2.3, create version 1.2.4.

Reviewer: Which endpoint?

Author: Fixed.

> It is not possible to submit new mutable data to an existing patch version such as 1.2.3. Instead, create a new patch version such as 1.2.4 with updated mutable data such as categories.

Reviewer 2: Could we be more specific? like:
```suggestion
This is an endpoint for asset creation, therefore if the asset already exists the request will fail with a 409 (Conflict), if you only want to modify the mutable data you will have to change the version of the asset, to publish a new one.
```

Maybe the last part is not necessary since we are going to have another endpoint to do that in the short future.

Author: Yes, that's better! We'll use this for now and update it when we have the other endpoint.

> This example publishes the category and custom field, which have the keys `categoryKey` and `fieldKey`. They must be configured for your organization. Publication fails if they are not configured for your organization.

Reviewer: redundant
```suggestion
This example publishes the category and custom field, which have the keys `categoryKey` and `fieldKey`.  Publication fails if they are not configured for your organization.
```
Do we need to clarify "they"?

Author: Good catch! Fixed both.

> The documentation archive file sent as `files.docs.zip` contains each page as a Markdown file such as `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page. Include resources such as images in the optional subdirectory `resources`. Specify the page order with the property `pageOrder` on the root of the main object in the optional file `config.json`:

Reviewer 2: ```suggestion
The documentation archive file sent as `files.docs.zip` contains each page as a Markdown file such as `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page. Include resources such as images in the optional subdirectory `resources`. To modify the terms and conditions page, the file `terms.md` must be present and its content will be the content of that section. Specify the page order with the property `pageOrder` on the root of the main object in the optional file `config.json`:
```
Could we add the `terms.md` section here? ☝️

Author: That's perfect! Added.

> The documentation archive file sent as `files.docs.zip` contains each page as a Markdown file such as `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page. To include a terms and conditions page, put the content in a file named `terms.md`. Include resources such as images in the optional subdirectory `resources`. Specify the page order with the property `pageOrder` on the root of the main object in the optional file `config.json`:

Reviewer: ```suggestion
The documentation archive file sent as `files.docs.zip` contains pages as Markdown files titled `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page. To include a terms and conditions page, put the content in a file named `terms.md`. Include resources such as images in the optional subdirectory `resources`. Specify the page order with the property `pageOrder` on the root of the main object in the optional file `config.json`:
```

Author: Fixed.

> The documentation archive file sent as `files.docs.zip` contains each page as a Markdown file such as `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page. To include a terms and conditions page, put the content in a file named `terms.md`. Include resources such as images in the optional subdirectory `resources`. Specify the page order with the property `pageOrder` on the root of the main object in the optional file `config.json`:

Reviewer: Clarify the repeated text "The documentation archive file sent as `files.docs.zip` contains each page as a Markdown file such as `Getting Started.md` in the root directory of the archive. Replace `Getting Started` with the title of the page."

Do you mean "as" rather than "such as"? Does every such Markdown file have the title Getting Started, until you change it?

Author: No, the user can use any title for the files, and "such as `Getting Started.md`" is intended to indicate that capitalization and spaces are supported, and the extension should be .md. I rewrote this for clarity.

[source,console]
----
{
 "pageOrder": ["home", "Getting Started", "Examples"]
}
----

[source,console]
----
curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/organizations/:organizationId/assets/:groupId/:assetId/:version' \
--header 'Authorization: bearer YOUR_TOKEN' \

> --header 'x-sync-publication: true' \

Reviewer 3: Can we add the contact name and contact email fields also here in the example?

Author: Good idea. Done!
